app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -236,10 +246,48 @@ public void addFromImage(View view) {
+        if (shouldScaleMedium) {
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedIcon.setVisibility(View.GONE);
+        } else {
+            int buttonMinHeight = getResources().getDimensionPixelSize(R.dimen.scan_button_min_height);
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedTitle.setVisibility(shouldScaleSmaller ? View.GONE : View.VISIBLE);
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedIcon.setVisibility(shouldScaleSmaller ? View.GONE : View.VISIBLE);
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedMessage.setTextSize(TypedValue.COMPLEX_UNIT_PX, getResources().getDimensionPixelSize(shouldScaleSmaller ? R.dimen.no_data_min_textSize : R.dimen.no_data_max_textSize));
+            customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedMessage.setPadding(shouldScaleSmaller?compatPadding:normalPadding,shouldScaleSmaller?compatPadding:normalPadding,shouldScaleSmaller?compatPadding:normalPadding,shouldScaleSmaller?compatPadding:normalPadding);
+            customBarcodeScannerBinding.addFromImage.setTextSize(TypedValue.COMPLEX_UNIT_PX, getResources().getDimensionPixelSize(shouldScaleSmaller ? R.dimen.scan_button_min_textSize : R.dimen.scan_button_max_textSize));
+            customBarcodeScannerBinding.addManually.setTextSize(TypedValue.COMPLEX_UNIT_PX, getResources().getDimensionPixelSize(shouldScaleSmaller ? R.dimen.scan_button_min_textSize : R.dimen.scan_button_max_textSize));
+            customBarcodeScannerBinding.addFromImage.setMinimumHeight(shouldScaleSmaller ? getResources().getDimensionPixelSize(R.dimen.scan_button_min_height) : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addFromImage.setMinimumWidth(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addManually.setMinimumHeight(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addManually.setMinimumWidth(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addFromImage.setMinHeight(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addFromImage.setMinWidth(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addManually.setMinHeight(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+            customBarcodeScannerBinding.addManually.setMinWidth(shouldScaleSmaller ? buttonMinHeight : buttonDefaultMinHeight);
+        }
+
+    }
 
+    private void saveDefaultUIValues() {
+        buttonDefaultMinHeight = customBarcodeScannerBinding.addFromImage.getMinHeight();
+        buttonDefaultMaxHeight = customBarcodeScannerBinding.addFromImage.getMinWidth();
     }
```

Comment:
Indeed, supporting different screen sizes is actually one of the headaches in Android. But no worries, I love doing research on things that I'm not familiar with because that's the only way how I can figure them out. At least now we know that there're boundaries for screen size support on Android.  

Hiding the title can provide more screen space and for now, it seems like the best solution that could work in most scenarios. So let's just use your suggested code.